tests/contracts/foundrylike/src/test_adder.rs

```diff
@@ -0,0 +1,91 @@
+#![no_std]
+
+pub mod testapi;
+
+multiversx_sc::imports!();
+
+static INIT_SUM : u32 = 5u32;
+#[multiversx_sc::contract]
+pub trait TestAdder {
+
+    #[storage_mapper("ownerAddress")]
+    fn owner_address(&self) -> SingleValueMapper<ManagedAddress>;
+
+    #[storage_mapper("adderAddress")]
+    fn adder_address(&self) -> SingleValueMapper<ManagedAddress>;
+
+    /// Create the owner account and deploy adder
+    #[init]
+    fn init(&self, code_path: ManagedBuffer) {
+        
+        // create the owner account
+        let owner = ManagedAddress::from(b"owner___________________________");
+        self.owner_address().set(&owner);
+        
+        testapi::create_account(&owner, 1, &BigUint::from(0u64));
+        
+        // register an address for the contract to be deployed
+        let adder = ManagedAddress::from(b"adder___________________________");
+        testapi::register_new_address(&owner, 1, &adder, );
+
+        // deploy the adder contract
+        let mut adder_init_args = ManagedArgBuffer::new();
+        adder_init_args.push_arg(INIT_SUM); // initial sum
+
+        // deploy a contract from `owner`
+        let adder = testapi::deploy_contract(
+                &owner,
+                5000000000000,
+                &BigUint::zero(),
+                &code_path,
```

Comment:
I've re-read your answers, and this seems to be a code path. The code may be easier to understand if, in testapi.rs, this would be called a "code path handle", not just a "code handle"